src/AspNetCoreModuleV2/AspNetCore/ServerErrorHandler.h

```diff
@@ -6,18 +6,95 @@
+                 <li> Enable logging the application process' stdout messages </li> \
+                 <li> Attach a debugger to the application process and inspect </li></ul></fieldset> \
+                 <fieldset><h4> For more information visit: \
+                 <a href=\"https://go.microsoft.com/fwlink/?LinkID=808681\"> <cite> https://go.microsoft.com/fwlink/?LinkID=808681 </cite></a></h4> \
+                 </fieldset> \
+              </div> \
+           </div></body></html>";
+
+    std::string ServerErrorHandler::s_html500PageOutOfProcess = "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Strict//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\"> \
+            <html xmlns=\"http://www.w3.org/1999/xhtml\"> \
+            <head> \
+            <meta http-equiv=\"Content-Type\" content=\"text/html; charset=iso-8859-1\" /> \
+            <title> IIS 500 Error </title><style type=\"text/css\"></style></head> \
+            <body> <div id = \"content\"> \
+              <div class = \"content-container\"><h3> HTTP Error 500 - ANCM OutOfProcess Startup Failure </h3></div>  \
+              <div class = \"content-container\"> \
+               <fieldset> <h4> Common causes of this issue: </h4> \
+                <ul><li>The out of process request handler, aspnetcorev2_outofprocess.dll, could not be found next to the aspnetcorev2.dll.</li> \
+                 <li> Could not read configuration correctly. Check the application's associated web.config.</li></ul></fieldset> \
+              </div> \
+              <div class = \"content-container\"> \
+                <fieldset><h4> Troubleshooting steps: </h4> \
+                 <ul><li> Check the system event log for error messages </li> \
+                 <li> Enable logging the application process' stdout messages </li> \
+                 <li> Attach a debugger to the application process and inspect </li></ul></fieldset> \
+                 <fieldset><h4> For more information visit: \
+                 <a href=\"https://go.microsoft.com/fwlink/?LinkID=808681\"> <cite> https://go.microsoft.com/fwlink/?LinkID=808681 </cite></a></h4> \
+                 </fieldset> \
+              </div> \
+           </div></body></html>";
+
 public:
-    ServerErrorHandler(IHttpContext &pContext, HRESULT hr) : m_pContext(pContext), m_HR(hr)
+    ServerErrorHandler(IHttpContext &pContext, HRESULT hr, APP_HOSTING_MODEL hostingModel, bool disableStartupPage)
+        : m_pContext(pContext), m_HR(hr), m_hostingModel(hostingModel), m_disableStartupPage(disableStartupPage)
     {
     }
 
     REQUEST_NOTIFICATION_STATUS OnExecuteRequestHandler() override
     {
-        m_pContext.GetResponse()->SetStatus(500, "Internal Server Error", 0, m_HR);
+        if (!m_disableStartupPage && m_hostingModel != HOSTING_UNKNOWN)
+        {
+            HTTP_DATA_CHUNK   DataChunk;
+            IHttpResponse* pResponse = m_pContext.GetResponse();
```

Comment:
We have a bunch of handlers that do this. Make a shared class you can pass text into.